include/exadg/incompressible_navier_stokes/spatial_discretization/operator_consistent_splitting.cpp

```diff
@@ -0,0 +1,776 @@
+  unsigned int dof_index_velocity  = this->get_dof_index_velocity();
+  unsigned int quad_index_pressure = this->get_quad_index_pressure();
+
+  FaceIntegratorP pressure_minus(matrix_free, true, dof_index_pressure, quad_index_pressure);
+  FaceIntegratorP pressure_plus(matrix_free, false, dof_index_pressure, quad_index_pressure);
+  FaceIntegratorU velocity_minus(matrix_free, true, dof_index_velocity, quad_index_pressure);
+  FaceIntegratorU velocity_plus(matrix_free, false, dof_index_velocity, quad_index_pressure);
+
+  for (unsigned int face = face_range.first; face < face_range.second; face++)
+  {
+    pressure_minus.reinit(face);
+    pressure_plus.reinit(face);
+    velocity_minus.reinit(face);
+    velocity_plus.reinit(face);
+
+    velocity_minus.gather_evaluate(src, dealii::EvaluationFlags::values);
+    velocity_plus.gather_evaluate(src, dealii::EvaluationFlags::values);
+
+    for (const unsigned int q : pressure_minus.quadrature_point_indices())
+      {
+        const auto normal = pressure_minus.normal_vector(q);
+        const auto div_factor =
+          - 0.5 * (velocity_minus.get_value(q) + velocity_plus.get_value(q)) * normal;
+
+        pressure_minus.submit_value(div_factor, q);
+        pressure_plus.submit_value(-div_factor, q);
+      }
+
+    pressure_minus.integrate_scatter(dealii::EvaluationFlags::values, dst);
+    pressure_plus.integrate_scatter(dealii::EvaluationFlags::values, dst);
+  }
+}
+
+
+template<int dim, typename Number>
+void
+OperatorConsistentSplitting<dim, Number>::compute_Leray_projection_boundary(
+  dealii::MatrixFree<dim, Number> const & matrix_free,
+  VectorType &                            dst,
+  VectorType const &                      src,
+  Range const &                           face_range) const
+{
+  unsigned int dof_index_velocity  = this->get_dof_index_velocity();
+  unsigned int dof_index_pressure  = this->get_dof_index_pressure();
+  unsigned int quad_index_pressure = this->get_quad_index_pressure();
+
+  FaceIntegratorP pressure_minus(matrix_free, true, dof_index_pressure, quad_index_pressure);
+  FaceIntegratorU velocity_minus(matrix_free, true, dof_index_velocity, quad_index_pressure);
+
+  for (unsigned int face = face_range.first; face < face_range.second; face++)
+    {
+      dealii::types::boundary_id const boundary_id = matrix_free.get_boundary_id(face);
+
+      BoundaryTypeU const boundary_type =
+        this->boundary_descriptor->velocity->get_boundary_type(boundary_id);
+
+  if(boundary_type == BoundaryTypeU::Dirichlet or boundary_type == BoundaryTypeU::DirichletCached)
+      {
+          pressure_minus.reinit(face);
+
+          for (const unsigned int q : pressure_minus.quadrature_point_indices())
+          {
+            // This term cancels with the boundary condition
+            pressure_minus.submit_value({}, q);
+          }
+
+        pressure_minus.integrate_scatter(dealii::EvaluationFlags::values, dst);
```

Comment:
can we get away with just not calling `pressure_minus.integrate_scatter()`? I think we had the same today in 
https://github.com/exadg/exadg/pull/792#discussion_r2478369193